### PR TITLE
ROE-392 adding validation to due diligence page

### PR DESCRIPTION
**Pull request by SVarallo**
Jira link: https://companieshouse.atlassian.net/browse/ROE-392

Work to add validation to the due diligence fields

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/DueDiligenceValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto.DUE_DILIGENCE_FIELD;
+
+@ExtendWith(MockitoExtension.class)
+class DueDiligenceValidatorTest {
+    private static final String CONTEXT = "12345";
```
Comment (mwestacott) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/DueDiligenceValidatorTest.java:23]: Suggest calling this constant `LOGGING_CONTEXT`. Just a bit more specific.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 2]
```diff
@@ -38,6 +40,9 @@ public Errors validate(OverseasEntitySubmissionDto overseasEntitySubmissionDto,
         if(Objects.nonNull(overseasEntitySubmissionDto.getOverseasEntityDueDiligence())) {
             overseasEntityDueDiligenceValidator.validate(overseasEntitySubmissionDto.getOverseasEntityDueDiligence(), errors, loggingContext);
         }
+        if(Objects.nonNull(overseasEntitySubmissionDto.getDueDiligence())) {
```
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:43]: This kind of leads in to the scenario when neither of the due diligence blocks are supplied, which is related to the story we talked about this morning (and the need for more ACs to specify the business rules to check) - https://companieshouse.atlassian.net/browse/ROE-1304.

I'd therefore be happy for that test to be omitted for the moment, but perhaps see what others say.

**Comment by markpit** [comment]
needs a PR description with link to the story

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:43]: yes, if it is to be covered by another story already in backlog then happy to leave as is for now.

**Review by markpit — APPROVED** [review]

**Review by mwestacott — APPROVED** [review]

**Pull request merged**